### PR TITLE
Fix cacerts file watch

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

fix https://github.com/istio/istio/issues/38366

### CODE REVIEW

File: pilot/pkg/bootstrap/istio_ca.go [hunk 1]
```diff
@@ -338,8 +322,7 @@ func (s *Server) handleCACertsFileWatch() {
 			}
 
 			if event.Op&fsnotify.Write == fsnotify.Write || event.Op&fsnotify.Create == fsnotify.Create {
-				valid := isValidCACertsFile(event.Name)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/istio_ca.go:341]: another way is to watch files one by one, but we have 4 files to watch. So that's why we use dir watch here.

Comment (ramaraochavali) [reply pilot/pkg/bootstrap/istio_ca.go:341]: I did not understand. What would be the `event.name` here?

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/istio_ca.go:341]: Take `/var/run/secrets/istio` directory as an examplem, it is mounted from istio rootcert configmap. 
When we update the configmap, the name can be `..data`, `..2022_04_13_09_13_52.574172247`

```
$ ls -al /var/run/secrets/istio
total 12
drwxrwxrwx 3 root root 4096 Apr 13 09:13 .
drwxr-xr-x 7 root root 4096 Apr 13 09:08 ..
drwxr-xr-x 2 root root 4096 Apr 13 09:13 ..2022_04_13_09_13_52.574172247
lrwxrwxrwx 1 root root   31 Apr 13 09:13 ..data -> ..2022_04_13_09_13_52.574172247
lrwxrwxrwx 1 root root   20 Apr 13 09:08 root-cert.pem -> ..data/root-cert.pem

```

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/istio_ca.go:341]: No event with name `root-cert.pem` come

Comment (ramaraochavali) [reply pilot/pkg/bootstrap/istio_ca.go:341]: I see

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**